### ISSUE TITLE
OpenRA Servers not freeing memory correctly

### ISSUE THREAD

**Issue by Zimeon-** [OpenRA/OpenRA#16515]
### Issue Summary
Dedicated servers hosted appear to not free up memory after games. Memory usage grows as games are played and the server is not rebooted. Based on estimates, it's roughly 100Mb/day with ~10games played each day.

#### System Information
- Linux cobalt 3.10.0-957.12.1.el7.x86_64 #1 SMP Mon Apr 29 14:59:59 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux | CentOS Linux release 7.6.1810 (Core)
- Mono 4.6.2-4.el7
- OpenRa version 20190314 compiled from source
- RA + Tiberian Dawn

#### Additional Information:
Server memory usage (available memory) past 6 months
https://imgur.com/a/dfg5rpw

**Comment by Zimeon-** [comment]
Memory usage seems to be more stable after upgrade to mono 5.20.

https://imgur.com/a/i1lHFK1

**Comment by pchote** [comment]
Stats from the EU official servers after updating to mono 5.20
```
Mem  -  mod  - port  - games
---------------------------------
679MB    ra     1300    211 games
565MB    ra     1329    146 games
425MB    ra     1330    108 games
188MB     ra     1318    72 games
196MB     cnc    1307    67 games
163MB     cnc    1308    43 games
90MB     cnc    1309    26 games
65MB     cnc    1310    7 games
45MB     d2k    1314    11 games
45MB     d2k    1315    12 games
65MB     d2k    1316    4 games
49MB     d2k    1317    3 games
245MB     ra     1302   50 games
204MB     ra     1303   11 games
212MB     ra     1301   12 games
188MB     ra     1304   15 games
286MB     ra     1311   71 games
```

Still leaking, but likely less than before.

**Comment by teinarss** [comment]
Looks like changing to a map with custom rules in the lobby makes the server eating more memory, went from 87mb to 165mb just by cycle through a couple of maps with custom rules.

![image](https://user-images.githubusercontent.com/19173272/62836363-427fcf00-bc62-11e9-8168-7958ba674fb2.png)

**Comment by pchote** [comment]
Fixed by #16944.
